javascript/2306-naming-a-company.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @param {string[]} ideas
+ * @return {number}
+ */
+
+// Time complexity: O(n^2 * m)
+// Space complexity: O(n * m)
+
+var distinctNames = function (ideas) {
+    const wordMap = new Map();
+    let res = 0;
+
+    ideas.forEach((word) => {
+        const ch = word.charAt(0);
+        const substr = word.slice(1);
+        if (!wordMap.has(ch)) {
+            wordMap.set(ch, new Set());
+        }
+        wordMap.get(ch).add(substr);
+    });
+
+    for (const [ch1, set1] of wordMap.entries()) {
+        for (const [ch2, set2] of wordMap.entries()) {
+            if (ch1 === ch2) {
+                continue;
+            }
+            let intersect = 0;
```

Comment:
Suggestion: move to helper function
```js
const intersect = getIntersect();
```

Reply:
Thanks! Updated the code

Reply:
Few other suggestions pending

Reply:
Updated code: now using for loop instead of forEach.